holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
> That does not seem extensible. If I want to declare a new Element type, which has one of these properties how would I tell is_distribution1D about that?

Well, nothing says you'll remember to override ``_1d`` when you define your new class! It is good to hear it is one of the only things like this - maybe it just needs a better name then?